### ISSUE TITLE
Login UI: CSS variable, JWT Token, AmplifyUser management

### ISSUE THREAD

**Issue by rejji** [aws-amplify/amplify-ui#4428]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Creat 

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

The login ui is invoked on click of a button from the webpage which is rendering an initial page available to everyone. I have used [theme provider](https://ui.docs.amplify.aws/react/connected-components/authenticator/customization#theme-provider-theme) and checked [css variables](https://ui.docs.amplify.aws/react/theming/css-variables). Can you suggest and possibly include an example fragment about how one can chose to limit the login ui to its modal-box dimensions only (the one that is dark colored) and put it as an overlay on top of existing webpage (better slide it down and let the cancel button slide it up). 

Thanks

### What's the expected behaviour?

Login-ui should consume space for its modal-box dimension only. This should be specifiable from Authenticator props. 

### Help us reproduce the bug!

Follow the steps given for [theme provider](https://ui.docs.amplify.aws/react/connected-components/authenticator/customization#theme-provider-theme).

### Code Snippet

```typescript
// Put your code below this line.
import React from 'react';
import { Authenticator, Theme, ThemeProvider, useTheme } from '@aws-amplify/ui-react';
import { Amplify } from 'aws-amplify';
import { useNavigate } from 'react-router-dom';
import awsExports from '../../aws-exports';

Amplify.configure(awsExports);
// import Navbar from '../Navbar/Navbar';

export default function Login2() {
  const navigate = useNavigate();

  const { tokens } = useTheme();
  const theme: Theme = {
    name: 'Auth Example Theme',
    tokens: {
      colors: {
        background: {
          primary: {
            value: tokens.colors.neutral['90'].value,
          },
          secondary: {
            value: tokens.colors.neutral['100'].value,
          },
        },
        font: {
          interactive: {
            value: tokens.colors.white.value,
          },
        },
        brand: {
          primary: {
            10: tokens.colors.teal['100'],
            80: tokens.colors.teal['40'],
            90: tokens.colors.teal['20'],
            100: tokens.colors.teal['10'],
          },
        },
      },
      components: {
        tabs: {
          item: {
            _focus: {
              color: {
                value: tokens.colors.white.value,
              },
            },
            _hover: {
              color: {
                value: tokens.colors.yellow['80'].value,
              },
            },
            _active: {
              color: {
                value: tokens.colors.white.value,
              },
            },
          },
        },
      },
    },
  };

  return (
    <ThemeProvider theme={theme}>
      <Authenticator>
        {({ signOut, user }) => (
          <main>
            <h1>Hello {user.username}</h1>
            <button type="button" onClick={() => { signOut(); navigate(-1); }}>Sign out</button>
          </main>
        )}
      </Authenticator>
    </ThemeProvider>
  );
}

```

### Additional information and screenshots

<img width="1400" alt="Screenshot 2023-09-11 at 6 58 46 AM" src="https://github.com/aws-amplify/amplify-ui/assets/15764395/cfb7a541-1d21-4a0d-9932-e33c1c4828bd">

**Comment by reesscot** [comment]
@rejji In your screenshot above the Authenticator is already constrained to itself. Can you explain what you are trying to accomplish?

**Comment by rejji** [comment]
I am looking to display _only_ the login-UI having darkened colour and be able to update its css properties like width, height, border. This is the modal box containing UI with signIn, CreateAccount buttons. This appears to be contained in the div data-amplify-container. Screenshot attached. 
And, I do not want to show the white screen around this login-UI. This UI ("https://localhost:3000/login2") is made to appear on the screen on press of a button on the previous root screen ("https://localhost:3000"). The portion that is showing as white screen should better be the one that belongs to previous root screen.

Once this is done, I was thinking to present this login-UI as sliding down instead of flashing up instantly. And similarly on the cancel, it should slide up instead of vanishing instantly.

How can this be done?

<img width="1397" alt="Screenshot 2023-09-13 at 11 29 52 AM" src="https://github.com/aws-amplify/amplify-ui/assets/15764395/962aacd6-9320-454f-8d51-6375b930557f">

**Comment by rejji** [comment]
Overlay part is done. Question is, for the login-UI, how to change the height, width, position using [css variables (https://ui.docs.amplify.aws/react/theming/css-variables)? This is for Login2 component shown above. Presently I have changed it using extra parent divs and flex for the css but wanted to check if "--amplify-components-authenticator-modal-height" and other variables are the right choice.


<img width="1403" alt="Screenshot 2023-09-13 at 3 37 53 PM" src="https://github.com/aws-amplify/amplify-ui/assets/15764395/c8fb3ea5-1b69-42cd-8571-8a5e4c9b0781">

**Comment by reesscot** [comment]
@rejji Did you figure out how to style the Authenticator the way you wanted to? based on your layout, I would probably turn off the modal variation to avoid having to modify the model CSS. See: https://ui.docs.amplify.aws/react/connected-components/authenticator/configuration#variation

As far as Auth management, Amplify is designed to fully manage the ID, Access and Refresh tokens for you. You shouldn't need to store them yourself. Do you have a particular use case that requires access to them?

**Comment by rejji** [comment]
I am using Auth itself for the token management. Rest of the things were done.

**Comment by rejji** [comment]
I am using Auth itself for the token management. Rest of the things were done.
